lib/node_modules/@stdlib/utils/parse-ndjson/lib/main.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+// MODULES //
+
+var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
+var isFunction = require( '@stdlib/assert/is-function' );
+var parseJSON = require( '@stdlib/utils/parse-json' );
+
+
+// MAIN //
+
+/**
+ * Parses a string as newline-delimited JSON (NDJSON).
+ *
+ * @param {string} str - input string
+ * @param {Function} [reviver] - transformation function applied to each line
+ * @throws {TypeError} first argument must be a string
+ * @throws {TypeError} reviver must be a function
+ * @returns {Array<(*|Error)>} array of parsed values or parse errors
+ *
+ * @example
+ * var obj = parseNDJSON( '{"name":"John"}\n{"name":"Doe"}' );
+ * // returns [{'name':'John'}, {'name':'Doe'}]
+ */
+function parseNDJSON( str, reviver ) {
+    if ( !isString( str ) ) {
+        throw new TypeError( 'invalid argument. First argument must be a string. Value: `%s`.', str);
+    }
+    if ( arguments.length > 1 && !isFunction( reviver ) ) {
+        throw new TypeError( 'invalid argument. Reviver argument must be a function.Value: `%s`.', str );
+    }
+
+    // checks if str contains whitespace only
+    if (/^\s*$/.test(str)){
+        return []
+    }
+
+    var lines = str.split( '\n' );
+    var out = [];
+    
+    for ( var i = 0; i < lines.length; i++ ) {
+        try {
```

Comment:
Why is this wrapped in a try/catch? In what scenario will `parseJSON` throw in this implementation?

Reply:
I realised try/catch here is unnecessary. I've removed it.